docs/fundamentals/flow/client.md

```diff
@@ -0,0 +1,283 @@
+```
+
+````
+````{tab} Processing documents
+
+```python
+from jina.types.request import Response
+
+
+def my_callback(rep: Response):
+    docs = resp.docs
+    ...  # process docs here
+```
+
+````
+
+In the example below, our Flow passes the message then prints the result when successful.
+If something goes wrong, it beeps. Finally, the result is written to output.txt.
+
+```python
+from jina import Flow, Client
+from docarray import Document
+
+
+def beep(*args):
+    # make a beep sound
+    import sys
+
+    sys.stdout.write('\a')
+
+
+with Flow().add() as f, open('output.txt', 'w') as fp:
+    client = Client(port=f.port_expose)
+    client.post(
+        '/',
+        Document(),
+        on_done=print,
+        on_error=beep,
+        on_always=lambda x: x.docs.save(fp),
+    )
+```
+
+## Returning results from .post()
+
+If no callback is provided, `client.post()` returns a flattened `DocumentArray` containing all Documents of all Requests.
+
+By setting `return_responses=True` when creating a Client, this behavior can be modified to return a list of Responses instead.
+
+If a callback is provided, no results will be returned.
+
+```{admonition} Danger
+:class: danger
+Not using a callback function and instead returning results can come with a **serious performance penalty**.
+
+Callbacks operate on each individual Request, which represents a batch of the data.
+In contrast, returning results requires the accumulation of all results of all Requests.
+This means that you will not receive results until all Requests have been processed.
+This may not only be slower, but also require more memory.
+```
+
+````{tab} Returning DocumentArray
+
+```python
+from jina import Flow, Client
+from docarray import Document
+
+with Flow() as f:
+    client = Client(port=f.port_expose)
+    docs = client.post(on='', inputs=Document(text='Hi there!'))
+    print(docs)
+    print(docs.texts)
+```
+```  
+>>> <DocumentArray (length=1) at 140619524357664>
+>>> ['Hi there!']
+```
+
+````
+````{tab} Returning Responses
+
+```python
+from jina import Flow, Client
+from docarray import Document
+
+with Flow() as f:
+    client = Client(port=f.port_expose, return_responses=True)
+    resp = client.post(on='', inputs=Document(text='Hi there!'))
+    print(resp)
+    print(resp[0].docs.texts)
+```
+``` 
+>>> [<jina.types.request.data.DataRequest ('header', 'parameters', 'routes', 'data') at 140619524354592>]
+>>> ['Hi there!']
+```
+
+````
+````{tab} Using callback function
+
+```python
+from jina import Flow, Client
+from docarray import Document
+
+with Flow() as f:
+    client = Client(port=f.port_expose)
+    resp = client.post(
+        on='',
+        inputs=Document(text='Hi there!'),
+        on_done=lambda resp: print(resp.docs.texts),
+    )
+    print(resp)
+```
+```
+>>> ['Hi there!']
+>>> None
+```
+
+````
+
+## Async Python Client
+
+There is also an async version of the Python Client so that it can easily be used from `asyncio` context:
```

Comment:
Explain that the `async_client` allows a `coroutine` as input and returns an `async_iterator`